library/std/src/sys/pal/uefi/time.rs

```diff
@@ -202,7 +219,11 @@ pub(crate) mod system_time_internal {
             dur.as_secs().checked_add_signed((-timezone as i64) * SECS_IN_MINUTE as i64).unwrap();
 
         // Convert to seconds since 1900-01-01-00:00:00 in timezone.
-        let Some(secs) = secs.checked_sub(TIMEZONE_DELTA) else { return None };
+        let Some(secs) = secs.checked_sub(TIMEZONE_DELTA) else {
+            let new_tz =
+                (secs / SECS_IN_MINUTE - if secs % SECS_IN_MINUTE == 0 { 0 } else { 1 }) as i16;
+            return Err(new_tz);
```

Comment:
Is there a usecase for `to_uefi` outside of `to_uefi_loose`? Otherwise you could update the timezone here instead of returning an error.